jerry-core/ecma/base/ecma-helpers.h

```diff
@@ -52,6 +52,33 @@
  */
 #define ECMA_SET_POINTER(field, non_compressed_pointer) MEM_CP_SET_POINTER (field, non_compressed_pointer)
 
+/**
+ * Convert ecma-string's contents to a cesu-8 string and put it into a buffer.
+ */
+#define ECMA_STRING_TO_UTF8_STRING(ecma_str_ptr, utf8_ptr) \
+  lit_utf8_size_t utf8_ptr ## _size; \
+  bool utf8_ptr ## _is_ascii; \
+  const lit_utf8_byte_t *utf8_ptr = ecma_string_raw_chars (ecma_str_ptr, &utf8_ptr ## _size, &utf8_ptr ## _is_ascii); \
+  bool utf8_ptr ## must_be_freed = false; \
```

Comment:
`utf8_prt ## _size` is a bad smell to me. I works, but does not look right. The variables `## _is_ascii` and `## must_be_freed` are also defined in the macro but are local, i.e., they are never used outside `ECMA_STRING_TO_UTF8_STRING` and `ECMA_FINALIZE_UTF8_STRING`. (Btw, why is it not `## _must_be_freed`? -- note the extra underscore.)

On the contrary, the macro defines `utf8_prt ## _size` to be used by the invoker, which is somewhat hidden, or not obvious from the signature of the macro. At first sight of the invocations of this macro, I've guessed that `ecma_str_ptr` is an input to the macro, `utf8_ptr` is something that gets defined by the macro, and that's it. Only then have I started wondering where the `_size` variables come from.

I'm not sure whether we have anything similar to this approach in the code base (i.e., a variable used outside of the macro that is not mentioned explicitly in the parameters). I'm not sure whether I have a much better alternative. I'd improve the docs of the macro(s) at the very least.